TicTacToe2.Tests/Vec2Tests.cs

```diff
@@ -34,7 +34,209 @@ public void TestAddX()
+                Assert.AreEqual(3, c.X);
+                Assert.AreEqual(4, c.Y);
+            }
+
+            {
+                var a = new Vec2(37, 5);
+                var b = new Vec2(-4, -8);
+                var c = a.Add(b);
+                Assert.AreEqual(33, c.X);
+                Assert.AreEqual(-3, c.Y);
+            }
+
+            {
+                var a = new Vec2(154, 663);
+                var b = new Vec2(0, 0);
+                var c = a.Add(b);
+                Assert.AreEqual(154, c.X);
+                Assert.AreEqual(663, c.Y);
+            }
+        }
+
+        [TestMethod]
+        public void TestSubtractX()
+        {
+            {
+                var a = new Vec2(6, 1);
+                var b = a.SubtractX(1);
+                Assert.AreEqual(5, b.X);
+                Assert.AreEqual(1, b.Y);
+            }
+
+            {
+                var a = new Vec2(37, 5);
+                var b = a.SubtractX(-41);
+                Assert.AreEqual(78, b.X);
+                Assert.AreEqual(5, b.Y);
+            }
+
+            {
+                var a = new Vec2(154, 663);
+                var b = a.SubtractX(0);
+                Assert.AreEqual(154, b.X);
+                Assert.AreEqual(663, b.Y);
+            }
+        }
+
+        [TestMethod]
+        public void TestSubtractY()
+        {
+            {
+                var a = new Vec2(1, 5);
+                var b = a.SubtractY(1);
+                Assert.AreEqual(1, b.X);
+                Assert.AreEqual(4, b.Y);
+            }
+
+            {
+                var a = new Vec2(37, 5);
+                var b = a.SubtractY(-41);
+                Assert.AreEqual(37, b.X);
+                Assert.AreEqual(46, b.Y);
+            }
+
+            {
+                var a = new Vec2(154, 663);
+                var b = a.SubtractY(0);
+                Assert.AreEqual(154, b.X);
+                Assert.AreEqual(663, b.Y);
+            }
+        }
+
+        [TestMethod]
+        public void TestSubtract()
+        {
+            {
+                var a = new Vec2(1, 1);
+                var b = a.Subtract(1, 1);
+                Assert.AreEqual(0, b.X);
+                Assert.AreEqual(0, b.Y);
+            }
+
+            {
+                var a = new Vec2(37, 5);
+                var b = a.Subtract(-41, -30);
+                Assert.AreEqual(78, b.X);
+                Assert.AreEqual(35, b.Y);
+            }
+
+            {
+                var a = new Vec2(154, 663);
+                var b = a.Subtract(0, 0);
+                Assert.AreEqual(154, b.X);
+                Assert.AreEqual(663, b.Y);
+            }
+
+        }
+
+        [TestMethod]
+        public void TestSubtract2()
+        {
+            {
+                var a = new Vec2(7, 15);
+                var b = new Vec2(1, 1);
+                var c = a.Subtract(b);
+                Assert.AreEqual(-6, c.X);
+                Assert.AreEqual(-14, c.Y);
+            }
```

Comment:
(7, 15) - (1, 1) should equal (positive) (6, 14). The below cases have the same issue.